src/main/java/Main.java

```diff
@@ -0,0 +1,117 @@
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import static org.apache.commons.io.FileUtils.readLines;
+
+public class Main {
+
+
+    public static void main(String[] args) {
+
+        String path;
+
+        if (args.length==0 || args[0].isEmpty()) {
+            Scanner input = new Scanner(System.in);
+            System.out.println("Enter input file name with full path to it : ");
+            path = input.nextLine();
+        }else {
+
+            path = args[0];
+        }
+        System.out.println(path);
+
+        ArrayList<String> rows = new ArrayList<>();
+        ArrayList<Integer> digits = new ArrayList<>();
+
+        //  TODO: use FileUtils from Apache Commons library (add with maven) to replace below loop
+        //  https://stackoverflow.com/a/4580340
+        try {
+            File inputFile = new File(path);
+//          *********************************************************************************
+//          CLARIFICATION NEEDED: I used  FileUtils.readLines and got List<String> lines
+//          But I have no idea which method use to divide this List to integer and String.
+            List<String>  lines = FileUtils.readLines(inputFile, StandardCharsets.UTF_8);
+            System.out.println("I used  FileUtils.readLines and got List<String> lines : " + lines);
+//          ***********************************************************************************
+
+            Scanner inputScanner = new Scanner(inputFile);
+            while ( inputScanner.hasNextLine()) {
+
+                if (inputScanner.hasNextInt()){
+                    int b =  inputScanner.nextInt();
+                    digits.add(b);
+
+                }else{
+                    String data = inputScanner.nextLine();
+                    rows.add(data);
+
+                }
+
+            }
+
+        } catch (java.io.IOException e) {
+            System.out.println("File not found");
+            e.printStackTrace();
+        }
+        //  DONE: google "for-loop", replace all the loops below with for-loop
+
+
+        //  DONE: move to separate method: int concatAllTextLines(List<String> lines)
+//      1. Все строки склеенные в одну
```

Comment:
This comment is useless now, because concatAllTextLines already tells what's happening

Reply:
Done